Add nonces to VDAF sharding syntax & Prio3 (#117)
cfrg/draft-irtf-cfrg-vdaf: draft-irtf-cfrg-vdaf.md

(\*) Indicates a change that breaks wire compatibility with the previous draft.

04:
* Change Prio3 spec to include the nonce in joint randomness derivation. This 
  mitigates a toy attack on robustness where a client submits the same set of 
  input shares under two distinct nonces and exploits the use of the same joint 
  randomness in both preparation phases. The mitigated attack has the same query 
  complexity as a birthday attack.

* Change VDAF syntax to include a nonce as an input in the sharding phase. This

>   accommodates the above change to Prio3. 

Reviewer: You can remove the change log update; we typically wait to do this until the next draft is cut.

FWIW, I would summarize the Prio3 as follows

> Prio3: Add the nonce to the transcript for joint randomness computation. This improves the provable security bound for robustness.

require interaction among the Aggregators in order to facilitate verifiability
of the computation's correctness. Accommodating this interaction will require
syntactic changes.

> One change is that in a VDAF, each client measurement is identified by a unique
> nonce provided by the environment (`nonce`, see {{run-vdaf}}).

Reviewer: I don't think this is all that useful here. 
```suggestion
```

## Sharding {#sec-vdaf-shard}

Sharding transforms a measurement into input shares as it does in DAFs 

> (cf. {{sec-daf-shard}}); in VDAFs it additionally takes a nonce as input and 

Reviewer: ```suggestion
(cf. {{sec-daf-shard}}); in addition, it takes a nonce as input and 
```

produces a public share:

* `Vdaf.measurement_to_input_shares(measurement: Measurement, nonce: Bytes) -> 
  (Bytes, Vec[Bytes])` is the randomized input-distribution algorithm run by each

>   Client. It consumes the measurement and the nonce and 

Reviewer: * `make` fails if a line ends with trailing white space:
  ```
  $ make
  757:Sharding transforms a measurement into input shares as it does in DAFs
  758:(cf. {{sec-daf-shard}}); in VDAFs it additionally takes a nonce as input and
  761:* `Vdaf.measurement_to_input_shares(measurement: Measurement, nonce: Bytes) ->
  763:  Client. It consumes the measurement and the nonce and
  draft-irtf-cfrg-vdaf.md contains trailing whitespace
  *** Run 'make fix-lint' to automatically fix some errors
  make: *** [lint-whitespace] Error 1
  ```

* Also, lines should break at 80 characters.

```suggestion
  Client. It consumes the measurement and the nonce and
```

~~~

> def measurement_to_input_shares(Poplar1, measurement, nonce):

Reviewer: ```suggestion
def measurement_to_input_shares(Poplar1, measurement, _nonce):
```
